File: backend/model/method.go
package model

import "github.com/google/uuid"

type MethodLabel string

const (
	METHOD_LABEL_SINGLE MethodLabel = "METHOD_LABEL_SINGLE"
	METHOD_LABEL_PAIR   MethodLabel = "METHOD_LABEL_PAIR"
	METHOD_LABEL_GROUP  MethodLabel = "METHOD_LABEL_GROUP"
	METHOD_LABEL_PLENUM MethodLabel = "METHOD_LABEL_PLENUM"
)

type Method struct {
	ID          uuid.UUID     `json:"id" gorm:"type:uuid;primary_key;"`
	Title       string        `json:"title"`
	Description string        `json:"description"`
	Labels      []MethodLabel `json:"labels" gorm:"type:text[]"`
}

File: frontend/src/types/method.ts
import { ContentDto } from "./content";

export enum Label {
  METHOD_LABEL_SINGLE = "METHOD_LABEL_SINGLE",
  METHOD_LABEL_PAIR = "METHOD_LABEL_PAIR",
  METHOD_LABEL_GROUP = "METHOD_LABEL_GROUP",
  METHOD_LABEL_PLENUM = "METHOD_LABEL_PLENUM",
}

export function resolveLabelName(label: Label): string {
  switch (label) {
    case Label.METHOD_LABEL_SINGLE:
      return "Single Person Working";
    case Label.METHOD_LABEL_PAIR:
      return "Partner Work";
    case Label.METHOD_LABEL_GROUP:
      return "Group Work";
    case Label.METHOD_LABEL_PLENUM:
      return "Plenum";
    default:
      return label;
  }
}

export interface MethodDto {
  id: string;
  title: string;
  description: string;
  category: string;
  labels: Array<Label>;
}

export interface BoardMethodDto extends MethodDto {
  ideas: Array<ContentDto>;
}

export interface CreateMethodDto {
  title: string;
  description: string;
  category: string;
  labels: Array<Label>;
}
